test/e2e/functional/tests/e2e.go

```
@@ -275,64 +293,112 @@ var _ = Describe("metallb", func() {
+
+				Eventually(func() error {
 					daemonset, err := testclient.Client.DaemonSets(metallb.Namespace).Get(context.Background(), consts.MetalLBDaemonsetName, metav1.GetOptions{})
 					if err != nil {
-						return false
+						return err
 					}
-					return daemonset.Status.DesiredNumberScheduled == daemonset.Status.NumberReady
-				}, metallbutils.DeployTimeout, metallbutils.Interval).Should(BeTrue())
 
-				pods, err = testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
-					LabelSelector: "component=speaker"})
-				Expect(err).ToNot(HaveOccurred())
-				daemonset, err := testclient.Client.DaemonSets(metallb.Namespace).Get(context.Background(), consts.MetalLBDaemonsetName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				Expect(len(pods.Items)).To(Equal(int(daemonset.Status.DesiredNumberScheduled)))
-				var speakerContainerFound bool
-				for _, pod := range pods.Items {
-					Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
-					for _, container := range pod.Spec.Containers {
-						if container.Name == "speaker" {
-							Expect(container.Resources).NotTo(BeNil())
-							Expect(container.Resources.Limits.Cpu().MilliValue()).To(Equal(int64(100)))
-							speakerContainerFound = true
+					if daemonset.Status.DesiredNumberScheduled != daemonset.Status.NumberReady {
+						return pkgerrors.New("speaker daemonset is not ready")
+					}
+
+					pods, err := testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
+						LabelSelector: "component=speaker"})
+					if err != nil {
+						return err
+					}
+
+					if len(pods.Items) != int(daemonset.Status.DesiredNumberScheduled) {
+						return pkgerrors.New("speaker daemonset pods are not ready")
+					}
+
+					var speakerContainerFound bool
+					for _, pod := range pods.Items {
+						Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
```

we should return an err here instead